src/domains/shielder/stores/getShielderIndexedDB.ts

```diff
@@ -60,7 +60,17 @@ type DBSchema = {
   },
 };
 
-const initDB = async (): Promise<IDBPDatabase<DBSchema>> => {
+const deleteDatabase = (): Promise<void> => {
+  const deleteRequest = indexedDB.deleteDatabase(DB_NAME);
+
+  return new Promise<void>((resolve, reject) => {
+    deleteRequest.onsuccess = () => void resolve();
+    deleteRequest.onerror = () => void reject(new Error(deleteRequest.error?.message ?? 'Database deletion failed'));
+    deleteRequest.onblocked = () => void reject(new Error('Database deletion blocked. Please close other tabs using this application.'));
+  });
```

Comment:
Doesn't the installed `idb` solve this?

--

If not, a nitpick, but I'd find it clearer if this whole thing was enclosed in that Promise:
```ts
const deleteDatabase = () => new Promise((resolve, reject) => {
  const deleteRequest = indexedDB.deleteDatabase(DB_NAME);
  deleteRequest.onsuccess = resolve;
  ...
```